cdap-docs/developers-manual/source/pipelines/developing-plugins/creating-a-plugin.rst

```diff
@@ -466,6 +482,13 @@ In order to implement a Streaming Source Plugin, you extend the ``StreamingSourc
   stream or dataset is not a macro.
 - ``getStream()``: Returns the ``JavaDStream`` that will be used as a source in the pipeline.
 
+.. rubric:: Lineage
+
+The lineage is registered using the ``referenceName`` provided when invoking ``registerLineage()`` on
+``StreamingContext`` in ``getStream()``. Note that the ``referenceName`` should be a valid ``DatasetId``.
+Optionally, you can extend from ``ReferenceStreamingSource``, which verifies the validity
```

Comment:
shouldn't mention this if we keep ReferenceStreamingSource where it is (which I think we need to do in order to avoid pulling in spark dependencies everywhere).